test/test_dht.cpp

```diff
@@ -288,7 +288,7 @@ void send_dht_request(node& node, char const* msg, udp::endpoint const& ep
 	if (ec) std::printf("bdecode failed: %s\n", ec.message().c_str());
 
 	dht::msg m(decoded, ep);
-	node.incoming(m);
+	node.incoming(node.m_sock, m);
```

Comment:
is this the only reason ``m_sock`` was made ``public``?

Reply:
Yes. Having made `m_sock` `const` I didn't see any harm in doing so.